### PR TITLE
(De-)serialize module settings

### LINKED ISSUE #91: Detail handling of module settings
Reviewing the settings concept, I noticed, that the *ModuleSettings classes contain no information as to which module they belong to.
This is currently missing, isn't it?

@Gottox Can you explain your idea behind the module settings handling?

Shouldn't there be a module identifier member in the AbstractModuleSettings class or something like this?

### PR DESCRIPTION
**Pull request by cburkert**
This provides the missing type adapters for (de-)serialization of module settings.

Closes #91

### CODE REVIEW

**Comment by cburkert** [comment]
I'll investigate the travis failure if I mange to get my build setup working again. It's totally broken after I moved to the meta repo...

**Comment by cburkert** [comment]
Problems fixed. I actually forgot to adapt the config test...
While fixing I noticed that the *ModuleSettings classes should be abstract, so I updated the code accordingly.

**Comment by thechauffeur** [comment]
@L-Henke if you have some time please review this PR.

File: src/test/java/de/qabel/core/config/ModuleSettingsTest.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+public class ModuleSettingsTest {
+	@Test
+	public void serializeModuleSettings() throws IOException {
+		SyncedModuleSettings fooSettings = new FooModuleSettings(42);
+		TypeAdapter<SyncedModuleSettings> adapter = new GsonBuilder()
+			.registerTypeAdapter(SyncedModuleSettings.class, new SyncedModuleSettingsTypeAdapter())
+			.create().getAdapter(SyncedModuleSettings.class);
+		String json = adapter.toJson(fooSettings);
+		System.out.println(json);
```
Comment (L-Henke) [inline src/test/java/de/qabel/core/config/ModuleSettingsTest.java]: It's just a test, but do you need the println?

Comment (cburkert) [inline src/test/java/de/qabel/core/config/ModuleSettingsTest.java]: Certainly not. I forgot to remove it.

**Pull request merged**